src/engine.rs

```diff
@@ -199,18 +217,17 @@ pub fn run_treefmt(
                             sum
                         });
                         // Return the new mtimes
-                        (formatter_name.clone(), new_paths)
+                        Ok((formatter_name.clone(), new_paths))
                     }
                     Err(err) => {
                         // FIXME: What is the right behaviour if a formatter has failed running?
-                        error!("{} failed: {}", &formatter, err);
                         // Assume the paths were not formatted
-                        (formatter_name.clone(), path_mtime.clone())
+                        return Err(anyhow!("{} failed: {}", &formatter, err));
                     }
                 }
             }
         })
-        .collect::<BTreeMap<FormatterName, BTreeMap<PathBuf, Mtime>>>();
+        .collect::<Result<BTreeMap<FormatterName, BTreeMap<PathBuf, Mtime>>, _>>()?;
```

Comment:
Is there no code later on that takes that into account?

If one formatter fails, the exit status of `treefmt` should also reflect that.